### PR TITLE
GSoD Vocabulary Landing Page (V1)

### PR DESCRIPTION
**Pull request by nimishbongale**
## Description
Finally! It's out! (Caution: Major PR with lots of changes! 😄)

This PR adds the new **Vocabulary Site v1** which was built as a part of my GSoD'20 work.

Find the VueJS site in the `docs` folder.

## Technical details

The site is currently live on [https://cc-vocab-draft.surge.sh](https://cc-vocab-draft.surge.sh)

## Screenshot

final-snap

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by kgodey** [comment]
Nice work @nimishbongale! @panchovm should probably review this too.

**Review by fcoveram — APPROVED** [review]

**Comment by fcoveram** [comment]
Thanks for this work @nimishbongale

**Comment by Dhruvi16** [comment]
This looks awesome @nimishbongale 🎊

**Review by Dhruvi16 — APPROVED** [review]

**Review by zackkrida — CHANGES_REQUESTED** [review]
Great work @nimishbongale, a few things:

- The `vocabulary, a monorepo?' section, please remove the pointer cursor styles that makes those numbered items seem like links.
- @panchovm The black sections use the regular font in all-caps, I haven't seen this anywhere else. Does this fit the style guide?
- I'd like to see a max-width (or the `.container` class) on the contents of the black and white sections. They get _very_ stretched on large screens:

- The images in the black sections overlap their containers; I'd rather see them with more padding

**Comment by nimishbongale** [comment]
Alright @zackkrida. I'll make the necessary changes. Thanks for reviewing! 😄

**Comment by nimishbongale** [comment]
> * I'd like to see a max-width (or the `.container` class) on the contents of the black and white sections. They get _very_ stretched on large screens:

@zackkrida 

I can set the max-width to some arbitrarily large value, but what do you think a suitable breakpoint should be?

Also, should I apply the `container` class to all the components we have? Some of them have it applied, while some don't.

**Comment by dhruvkb** [comment]
@nimishbongale the `.container` class itself has the max-width set for various breakpoints.

**Comment by nimishbongale** [comment]
Perfect. Sounds good!

**Comment by nimishbongale** [comment]
Is this (roughly) the expected behaviour at large screen sizes?

screencapture-localhost-8080-2020-10-30-17_02_28

(The site has also been redeployed to view: [https://cc-vocab-draft.surge.sh](https://cc-vocab-draft.surge.sh) )

**Comment by fcoveram** [comment]
Yes! @zackkrida thanks for noticing it. There is a mix of headings using different styles. `H1` and `H2` must use the Heading 1A and Heading 2A, respectively.

**Comment by zackkrida** [comment]
@nimishbongale i would try using the container class on a div inside the black backgrounds, so the black bg will be 100% wide but the content will be constrained.

**Comment by nimishbongale** [comment]
@zackkrida I'm not 100% sure about this, but does a `notification-container` apply similar styles as a normal `container`?

https://github.com/creativecommons/vocabulary/blob/8658041bd2cfe34fae29865d8c42687a10f54d47/docs/src/components/SectionTitle.vue#L3

**Comment by zackkrida** [comment]
@nimishbongale You're using the notification classes in a way they're not really intended to be used. They're not well-suited for page layout. I would suggest something more like this:

https://codesandbox.io/s/ecstatic-ganguly-5t2g9?file=/src/App.vue



You can learn more about bulma's _column_ classes here: https://bulma.io/documentation/columns/options/

**Comment by nimishbongale** [comment]
@zackkrida 

Ahha. Okay! I just looked for the most appropriate component! 😄 

Thank you!

**Comment by zackkrida** [comment]
@nimishbongale I'm not the biggest bulma fan—it's hard to tell what best practices are from their docs! No worries 😅

**Comment by nimishbongale** [comment]
@zackkrida 

I hope I've got it right this time! 😄

There's still some other work I've found that needs to be done (image assets cleanup, for starters). I'll keep pushing changes as & when they get reviewed.

(And ooh, we need to add some CI scripts for this site as well 😅 for now I'm just locally redeploying on [https://cc-vocab-draft.surge.sh](https://cc-vocab-draft.surge.sh))

**Comment by zackkrida** [comment]
Hi @nimishbongale, apologies for the delay in fully reviewing this! Anytime we create a new web property, there's a little more thought put into reviewing content, audience, and design. We'd like to simplify this site significantly. Here are some notes:

## Design + Copy Changes

- [x] Hero Section Edits:
  - [x] Switch the hero section to match the styles on https://opensource.creativecommons.org/ you can skip the image: just have a plain white background
  - [x] Replace "Design, Develop & Build with Creative Commons' very own design system!" with "Vocabulary is a cohesive design system and component library used on all Creative Commons web properties. The code is freely available to anyone and everyone to inspect, modify & use for their own purposes."
  - [x] Add an installation code snippet `<p><code class="body-big margin-top-bigger is-inline-block">npm install @creativecommons/vocabulary</code></p>` below the new copy
  - Switch to two buttons:
	  - [x] Go to Storybook (the unified one)
  	- [x] Go to GitHub
- [x] Remove "Our Mission" block, replace it with the 'package-group' section (that has a block for each package and "Visit npmjs" links)
- [x] Give the "Why Vocabulary?" section the black background styles
- [x] Move the stats blocks below this new header section
- [x] Remove all other sections (sorry about this, I know you worked hard on the release carousel!)
- [x] Change the navigation from jump links to the following external links:
  - [x] [Release Notes](https://github.com/creativecommons/vocabulary/releases)
  - [x] [Documentation](https://cc-vocabulary.netlify.app/)
  - [x] [GitHub](https://github.com/creativecommons/vocabulary)
- [x] Fix "Github" to "GitHub" in footer

**Comment by nimishbongale** [comment]
Okay that's a lot of changes haha! 😄 Although I did expect them to be fair 🤔 

I did think the release carousel was a neat little thing to have. But yeah, maybe not! I'll get to working on them right away! 

Thanks for the review @zackkrida 👍

**Comment by nimishbongale** [comment]
> Change the navigation from jump links to the following external links:
 Release Notes
 Documentation
 GitHub

In the footer? Or in the header? 

> Add an installation code snippet <p><code class="body-big margin-top-bigger is-inline-block">npm install @creativecommons/vocabulary</code></p> below the new copy

My only concern here is that it would mean we call for installation of vocabulary, and not the other packages. Should I add this step for all the three?

**Comment by nimishbongale** [comment]
Here's how it looks now:

(I haven't committed/deployed anything yet)

screencapture-localhost-8080-2020-11-13-23_02_41

I'm yet to add some colors (I think the site looks bland as is, definitely could do with some color). I'm just afraid the site has too less content to show. But maybe the simpler the better!

**Comment by zackkrida** [comment]
@nimishbongale I think it's great progress. 

- [x] Add the word "GitHub" before the GitHub logo in the header menu
- [x] Here is a header mockup I made, to add some color and consistency:
Screen Shot 2020-11-13 at 8 14 12 AM

- [x] Please remove the black monorepo section and add something like this above the 3 packages: 
Screen Shot 2020-11-13 at 8 20 35 AM


Those additions might provide enough color, for now! Thank you.

File: docs/src/components/Hero.vue [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  Nested Components: None
+-->
+
+<template>
+    <div style="min-height: 40vmax;">
+      <div :id="id" class="has-background-white padding-bigger section-title">
+      <div class="container">
+        <div class="content-wrap section-margin">
+          <div class="is-three-quarters">
+            <p class="title is-1 has-text-gray">WELCOME TO THE DESIGN SYSTEM OF CREATIVE COMMONS OPEN SOURCE.</p><br>
```
Comment (zackkrida) [inline docs/src/components/Hero.vue]: One note: You should never write text all-caps in HTML, always use `text-transform: uppercase`. There's an important reason for this: Some screen readers will read all-caps as acronyms! So every. single. letter. will be read out loud.

File: docs/src/components/Hero.vue [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            <p class="subtitle is-4" style="font-weight: normal; color: #767676;">{{ content }}</p><br>
+              <form action="https://cc-vocabulary.netlify.app" target="_blank">
+                <button class="button">VISIT DOCUMENTATION<center><br><i class="icon external-link"></i></center></button>
+              </form>
+              <br>
+              <form action="https://github.com/creativecommons/vocabulary" target="_blank">
+                <button class="button">VISIT GITHUB<center><br><i class="icon github"></i></center></button>
```
Comment (zackkrida) [inline docs/src/components/Hero.vue]: One note: You should never write text all-caps in HTML, always use `text-transform: uppercase`. There's an important reason for this: Some screen readers will read all-caps as acronyms! So every. single. letter. will be read out loud.

Comment (nimishbongale) [reply docs/src/components/Hero.vue]: Ahha. Cool! I'll fix. Thanks!

File: docs/src/components/Hero.vue (hunk 1 above)
Comment (nimishbongale) [reply docs/src/components/Hero.vue]: Turns out the `title` class applies it by default. Wonder why I was inserting text in caps 🤔

**Comment by nimishbongale** [comment]
@zackkrida 

How does this look now?

I've redeployed on [https://cc-vocab-draft.surge.sh](https://cc-vocab-draft.surge.sh) as well.

screencapture-cc-vocab-draft-surge-sh-2020-11-14-10_55_28

**Comment by zackkrida** [comment]
@nimishbongale it looks **great**! Thanks for iterating on the design so quickly. 

One last thing... please add the `small` class to the buttons in the hero, so they match the os site, and add a max-width to the 'subtitle' block in the hero, something like this:

```css
width: 500px;
max-width: 100%;
```

**Comment by nimishbongale** [comment]
@zackkrida 

I've added the necessary styles. I guess we could do with some sort of an image on the right hand side? I don't really know, it's just a thought! 😄 

screencapture-cc-vocab-draft-web-app-2020-11-16-17_19_02

**Comment by zackkrida** [comment]
> I guess we could do with some sort of an image on the right hand side? 

Eventually, yes, we'd just want it to really match the cc branding, which can be hard to come up with, and we don't have the resources for our designer to design something right now.

**Review by zackkrida — APPROVED** [review]

**Comment by kgodey** [comment]
I recommend removing the Why Vocabulary section, it takes up a lot of space and doesn't add much information. All CC projects are required to use Vocabulary for brand consistency so it's not really a choice...

**Comment by zackkrida** [comment]
@kgodey Agreed, but I think we still need a section design-wise to break up the package cards and the stats cards. Perhaps we just deemphasize the section; something like this?

cc-vocab-draft surge sh_

Also @nimishbongale please make sure the 'multiple ways to use' section has a `container` inside of it, it looks like it's contents expand to full-width right now.

**Comment by nimishbongale** [comment]
Here's how it looks now. I've redeployed it as well.

screencapture-cc-vocab-draft-surge-sh-2020-11-18-22_09_15

**Comment by kgodey** [comment]
Looks good, I would reduce the whitespace in the Why Vocabulary section similar to @zackkrida's mockup.

**Comment by nimishbongale** [comment]
> Looks good, I would reduce the whitespace in the Why Vocabulary section similar to @zackkrida's mockup.

Done! I've redeployed it, do have a look! 😄

- https://cc-vocab-draft.surge.sh

**Comment by zackkrida** [comment]
It's merged! 🎊 
Thank you so much @nimishbongale — I will let you know when i've deployed and set things up on Netlify.

**Comment by nimishbongale** [comment]
🎊 🎊

@dhruvkb @zackkrida @kgodey @panchovm @Dhruvi16 @obulat 

A big thank you to everyone for all their support and guidance! 👏  

It's for sure a wonderful feeling! 😄

**Comment by fcoveram** [comment]
Thank you @nimishbongale 🙌

**Pull request merged**